### PR TITLE
PR fix 2

### PR DESCRIPTION
**Pull request by felixfavour**

### CODE REVIEW

File: dataservice.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// function to load products JSON and convert it to a map.
+let getProducts = () => {
+    const productsJson = require('./my-app-react/src/data/products.json');
+    if (productMap.size === 0) {
+        for (let product of productsJson.products) {
+            productMap.set(product.id, product);
+        }
+    }
+    return productsJson
```
Comment (f4ww4z) [inline dataservice.js]: Should return the map, not original data.
```suggestion
    return productMap
```

File: dataservice.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+// function to load categories JSON and convert it to a map.
+let getCategories = () => {
+    const categoriesJson = require('./my-app-react/src/data/categories.json');
+    if (categoryMap.size === 0) {
+        for (let category of categoriesJson.categories) {
+            categoryMap.set(category.id, category.categoryName)
+        }
+    }
+    return categoriesJson
```
Comment (f4ww4z) [inline dataservice.js]: Above comment is similar here.

File: dataservice.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+// Function to combine both products and categories in a collectable and renderable format
+let combineProductsWithCategories = () => {
+    getProducts();
```
Comment (f4ww4z) [inline dataservice.js]: Return value is not used. Either don't return anything in the function or use it here.

File: dataservice.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    getCategories();
```
Comment (f4ww4z) [inline dataservice.js]: See above comment.

File: dataservice.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    for (let product of getProducts().products) {
```
Comment (f4ww4z) [inline dataservice.js]: You need to loop through `productMap`. Hint: Use `Object.values()`. Then, add `categoryName` attribute to produce a `combinedProductMap`

File: dataservice.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                product.categoryName = category.categoryName
+            }
+        }
+        combinedProductMap.set(product.id, product);
+    }
+    
+};
+// Function to return the result of above function
+exports.getCombinedProductMap = () => {
+    if(combinedProductMap.size === 0) {
```
Comment (f4ww4z) [inline dataservice.js]: What if `combinedProductMap` is undefined or null?

Comment (f4ww4z) [reply dataservice.js]: Have you also reformatted this? 
```suggestion
    if (combinedProductMap.size === 0) {
```

File: server.js [hunk 7]
```diff
@@ -1,49 +1,45 @@
...
-const express = require('express')
-const app = express()
+const express = require('express');
+const app = express();
 
 app.get('/info', (req, res) => {
-    info = [
-        {"serverName": "localhost5000"},
+    let info = [
+        {"serverName": "localhost5001"},
```
Comment (f4ww4z) [inline server.js]: Remove `sampleName`, and this should have the value `"chinemerem-react"`

File: server.js [hunk 8]
```diff
@@ -1,49 +1,45 @@
...
+});
 
 app.get('/products/all', (req, res) => {
-    res.json(products)
-})
+    let productsMap = dataService.getCombinedProductMap();
+    let allProducts = [];
+    productsMap.forEach(value => {
+       allProducts.push(value)
+    });
```
Comment (f4ww4z) [inline server.js]: No need for [these lines](https://github.com/felixfavour/my-react-app/pull/10/files#diff-78c12f5adc1848d13b1c6f07055d996eR17-R20), conversion not required. Simply return the map.

File: server.js [hunk 9]
```diff
@@ -1,49 +1,45 @@
...
-
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlProductID == product.id) {
-            res.json(product)
-        }
-    }
-
-})
+    let urlProductID = req.params.id.toUpperCase();
```
Comment (f4ww4z) [inline server.js]: Why to upper case?

File: server.js [hunk 10]
```diff
@@ -1,49 +1,45 @@
...
-    let productsLocal = products.products
-    let arrayOfProducts = []
-
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlCategoryID == product.categoryId) {
-            arrayOfProducts.push(product)
+    let urlCategoryID = req.params.id;
+    let combinedProductMap = dataService.getCombinedProductMap();
+    let allProducts = [];
```
Comment (f4ww4z) [inline server.js]: Should be a map, just like `combinedProductMap` but only contains those who have matching `ctyId`. Also, rename this variable.

File: server.js [hunk 11]
```diff
@@ -1,49 +1,45 @@
...
+    combinedProductMap.forEach(((value, key, map) =>{
```
Comment (f4ww4z) [inline server.js]: Run eslint again.
```suggestion
    combinedProductMap.forEach(((value, key, map) => {
```

File: server.js (hunk 8 above)
Comment (felixfavour) [reply server.js]: apparently, A map is rendered in the page as `{ }`

**Pull request closed without merging**